lbrynet/core/client/DHTPeerFinder.py

```diff
@@ -34,7 +38,17 @@ def _manage_peers(self):
         pass
 
     @defer.inlineCallbacks
-    def find_peers_for_blob(self, blob_hash, timeout=None):
+    def find_peers_for_blob(self, blob_hash, timeout=None, filter_self=False):
+        """
+        Find peers for blob in the DHT
+        blob_hash (str): blob hash to look for
+        timeout (int): seconds to timeout after
+        filter_self (bool): if True, and if a peer for a blob is itself, filter it
+                from the result
```

Comment:
(assume that I have not dug yet in the downloader code) I am wondering if knowing that `self` already has the blob could be a useful information to avoid downloading it again. Or is this case handled differently by the downloader (i.e. the downloader checks the blob directory directly before issuing a request)?

Reply:
There is a BlobManager class which knows all the blobs  it has. In theory , when the daemon is downloading it should not try to download blobs it has announced. But I think in practice, this can happen if for example you announce a blob, and than you delete the blob in blob manager and try to download the same blob again.

Reply:
OK, it makes sense. Thanks for clarifying!